pkg/trace/writer/trace_nix_test.go

```diff
@@ -51,141 +52,121 @@ func TestPeriodicFlush(t *testing.T) {
 		randomTracePackage(2, 0),
 	}
 
-	expectedMinNumRetries++
-	expectedNumTraces += 3
-	expectedNumSpans += 6
-	expectedNumBytes += calculateTracePayloadSize(payload4SampledTraces)
-
-	for _, sampledTrace := range payload4SampledTraces {
+	for _, sampledTrace := range payload4 {
 		traceChannel <- sampledTrace
 	}
 
-	// Wait for twice the flush period to see at least one retry
 	time.Sleep(2 * testFlushPeriod)
 
+	want["retries"] = 2
+	want["traces"] += len(payload1)
+	want["spans"] += 6
+	want["bytes_estimated"] += calculateTracePayloadEstimatedSize(payload4)
+	want["bytes_uncompressed"] += calculateTracePayloadSize(payload4)
+
 	// Close and stop
 	close(traceChannel)
 	traceWriter.Stop()
 
-	// Then we expect some counts to have been sent to the stats client for each update tick (there should have been
-	// at least 3 ticks)
-	countSummaries := statsClient.GetCountSummaries()
-
-	// Payload counts
-	payloadSummary := countSummaries["datadog.trace_agent.trace_writer.payloads"]
-	assert.True(len(payloadSummary.Calls) >= 3, "There should have been multiple payload count calls")
-	assert.Equal(expectedNumPayloads, payloadSummary.Sum)
-
-	// Traces counts
-	tracesSummary := countSummaries["datadog.trace_agent.trace_writer.traces"]
-	assert.True(len(tracesSummary.Calls) >= 3, "There should have been multiple traces count calls")
-	assert.Equal(expectedNumTraces, tracesSummary.Sum)
-
-	// Spans counts
-	spansSummary := countSummaries["datadog.trace_agent.trace_writer.spans"]
-	assert.True(len(spansSummary.Calls) >= 3, "There should have been multiple spans count calls")
-	assert.Equal(expectedNumSpans, spansSummary.Sum)
-
-	// Bytes counts
-	bytesSummary := countSummaries["datadog.trace_agent.trace_writer.bytes"]
-	assert.True(len(bytesSummary.Calls) >= 3, "There should have been multiple bytes count calls")
-	// FIXME: Is GZIP non-deterministic? Why won't equal work here?
-	assert.True(math.Abs(float64(expectedNumBytes-bytesSummary.Sum)) < 100., "Bytes should be within expectations")
-
-	// Retry counts
-	retriesSummary := countSummaries["datadog.trace_agent.trace_writer.retries"]
-	assert.True(len(retriesSummary.Calls) >= 2, "There should have been multiple retries count calls")
-	assert.True(retriesSummary.Sum >= expectedMinNumRetries)
-
-	// Error counts
-	errorsSummary := countSummaries["datadog.trace_agent.trace_writer.errors"]
-	assert.True(len(errorsSummary.Calls) >= 3, "There should have been multiple errors count calls")
-	assert.Equal(expectedNumErrors, errorsSummary.Sum)
-
-	// Single trace max spans
-	singleMaxSpansSummary := countSummaries["datadog.trace_agent.trace_writer.single_max_spans"]
-	assert.True(len(singleMaxSpansSummary.Calls) >= 3, "There should have been multiple single max spans count calls")
-	assert.Equal(expectedNumSingleMaxSpans, singleMaxSpansSummary.Sum)
+	expectStats := summaryCompareFunc(assert, statsClient)
+	for key, sum := range want {
+		delta := 1.
+		if key == "bytes_uncompressed" {
+			delta = 1000.
```

Comment:
How come we need such a big delta for uncompressed?

Reply:
This was a leftover from a previous draft implementation. It's good that you've noticed. I've completely removed it.